src/app/tracing/tracing.effects.ts

```diff
@@ -305,4 +308,57 @@ export class TracingEffects {
             return EMPTY;
         })
     ));
+
+    setStationPositionsAndLayoutMSA$ = createEffect(() => this.actions$.pipe(
+        ofType<tracingEffectActions.SetStationPositionsAndLayoutMSA>(
+            tracingEffectActions.TracingActionTypes.SetStationPositionsAndLayoutMSA
+        ),
+        withLatestFrom(this.store.pipe(select(tracingSelectors.getFclData))),
+        mergeMap(([action, state]) => {
+            try {
+                const oldStationPositionsAreEmpty = _.isEmpty(state.graphSettings.stationPositions);
+                const payloadPositionsAreEmpty = _.isEmpty(action.payload.stationPositions);
+
+                if ((oldStationPositionsAreEmpty) && (!payloadPositionsAreEmpty)) {
+                    new tracingStateActions.SetStationPositionsAndLayoutSOA(action.payload);
+
+                    const newFclData = {
+                        ...state,
+                        graphSettings: {
+                            ...state.graphSettings,
+                            stationPositions: action.payload.stationPositions,
+                            schemaLayout: action.payload.layout
+                        }
+                    };
+
+                    return of(
+                        new tracingStateActions.SetStationPositionsAndLayoutSOA(action.payload),
+                        new tracingEffectActions.RefreshReducedReferenceCopyMSA({ fclData: newFclData })
+                    );
+                } else {
+                    return of(new tracingStateActions.SetStationPositionsAndLayoutSOA(action.payload));
+                }
+            } catch (error) {
+                this.alertService.error(`station positions and layout could not be set!, error: ${error}`);
+            }
+            return EMPTY;
+        })
+    ));
```

Comment:
Instead of 
`));`
you need to write 
`), { dispatch: false });`
in case you are dispatching the actions yourself.

Reply:
done!